### PR TITLE
feat: leave MLS conference call after a crash

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

If we crashed out of an MLS conference call we should attempt to leave it after a restart. Implementing [Use case: leave conference calls after a crash](https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/699170905/Use+case+leave+conference+calls+after+a+crash+MLS)

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/mapper/CallMapper.kt [hunk 1]
```diff
@@ -157,6 +164,13 @@ class CallMapperImpl(
         else -> ConversationEntity.Type.ONE_ON_ONE
     }
 
+    private fun toCallEntityType(conversationType: ConversationType): CallEntity.Type = when (conversationType) {
+        ConversationType.OneOnOne -> CallEntity.Type.ONE_ON_ONE
+        ConversationType.Conference -> CallEntity.Type.CONFERENCE
+        ConversationType.ConferenceMls -> CallEntity.Type.MLS_CONFERENCE
+        ConversationType.Unknown -> CallEntity.Type.CONFERENCE
```
Comment (alexandreferris) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/mapper/CallMapper.kt:171]: Shouldn't here be `CallEntity.Type.UNKNOWN`?

**Review by alexandreferris — COMMENTED** [review]
looking dope 👌🏻  Just have a small comment on the mapper

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/mapper/CallMapper.kt (hunk 1 above)
Comment (typfel) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/mapper/CallMapper.kt:171]: Yes! typo

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt [hunk 2]
```diff
@@ -471,6 +477,7 @@ internal class CallDataSource(
         )
 
     override suspend fun updateOpenCallsToClosedStatus() {
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt:479]:
```suggestion
    override suspend fun closeOpenCalls() {
```

**Comment by typfel** [comment]
@alexandreferris during some manual testing I discovered that we weren't passing the `convType` argument when calling `wcall_receive_msg`.

**Comment by alexandreferris** [comment]
> @alexandreferris during some manual testing I discovered that we weren't passing the `convType` argument when calling `wcall_receive_msg`.

Hmmmm.. we based on existing implementation of calling interface from old Scala app, [can be seen here](https://github.com/wireapp/wire-android/blob/a3be24c143276cda6a5eec956ebe715d9be5ce15/zmessaging/src/main/scala/com/waz/service/call/Calling.scala), where did you find that `wcall_recv_msg` also expects a `convType`?

**Comment by typfel** [comment]
> > @alexandreferris during some manual testing I discovered that we weren't passing the `convType` argument when calling `wcall_receive_msg`.
> 
> Hmmmm.. we based on existing implementation of calling interface from old Scala app, [can be seen here](https://github.com/wireapp/wire-android/blob/a3be24c143276cda6a5eec956ebe715d9be5ce15/zmessaging/src/main/scala/com/waz/service/call/Calling.scala), where did you find that `wcall_recv_msg` also expects a `convType`?

It was introduced in AVS 9.x https://github.com/wireapp/wire-avs/blob/main/docs/api/api_changes_9.0.txt I guess more an over sight on my side since it was added to support the new call type  `MLS_CONFERENCE` calls.

**Review by alexandreferris — APPROVED** [review]

**Pull request merged**